_posts/en/newsletters/2021-01-13-newsletter.md

```diff
@@ -0,0 +1,217 @@
+  these *bech32m* addresses will be detected.  If no problems are found
+  with the proposal, it is expected that bech32m addresses will be used
+  for [taproot][topic taproot] addresses and possibly future new script
+  upgrades.  This will allow wallets and services that adopt bech32m to
+  automatically be able to pay future wallets and services that upgrade
+  to taproot and many other proposed improvements (see [Newsletter
+  #45][news45 bech32 upgrade] for details).
+
+- **LN dual funding anti UTXO probing:** a long-term goal for LN has
+  been dual funding, the ability to open a new channel with funds from
+  both the node initiating the channel and the peer receiving their
+  request.  This will allow payments to travel in either direction in
+  the channel from the moment it's fully opened.  Before the initiator
+  can sign the dual funding transaction, they need the identities
+  (outpoints) of all of the UTXOs the other party wants to add to the
+  transaction.  This creates a risk that an abuser will attempt to
+  initiate dual-funded channels with many different users, learn their
+  UTXOs, and then refuse to sign the funding transaction---harming those
+  users' privacy at no cost to the abuser.
+
+    This week, Lloyd Fournier [posted][fournier podle] to the Lightning-Dev
+    mailing list an evaluation of two previous proposals to deal
+    with this problem, [one][zmn podle] using Proofs of Discrete Log
+    Equivalency (PoDLEs, see [Newsletter #83][news83 podle]) and the
+    [other][darosior sighash] using dual funding transactions
+    half-signed with `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY`.  Fournier
+    extended the previous half-signed proposal and then provided his own
+    proposal that was equivalently effective but simpler.  The new
+    proposal has the initiator create and sign (but not broadcast) a
+    transaction that spends their UTXO back to themselves.  They give
+    this to the other party as a proof of good faith---if the initiator
+    later fails to sign the actual funding transaction, the respondent
+    can broadcast the good-faith transaction, forcing the initiator to
+    pay an onchain fee.  Fournier concludes his post with a summary of
+    the tradeoffs between the different approaches.
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review Club][]
+meeting, highlighting some of the important questions and answers.  Click on a
+question below to see a summary of the answer from the meeting.*
+
+[Add unit testing of node eviction logic][review club
+#20477] is a PR ([#20477][Bitcoin Core #20477]) by
+pseudonymous contributor practicalswift to improve test coverage of Bitcoin Core's peer
+eviction logic when a node's inbound connection slots are full. Extreme care
+must be taken to avoid exposing the node to attacker-triggered network
+partitioning with this logic, so this code is critical.
```

Comment:
I'm not sure about this sentence. Our outbound connection logic is much more important for partition resistance.

Reply:
It's from https://github.com/bitcoin/bitcoin/blob/7b975639ef93b50537a3ec6326b54d7218afc8da/src/net.cpp#L969, though not necessarily viable content in this context.￼￼